File: 0189-rotate-array/0189-rotate-array.cs
public class Solution {
    private void Swap(int[] nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
    public void Rotate(int[] nums, int k) 
    {
        int n = nums.Count();
        k %= n;
        Array.Reverse(nums);

        int i = 0, j = k-1;
        while (i < j) {
            Swap(nums, i++, j--);
        }

        i = k; j = n-1;
        while (i < j) {
            Swap(nums, i++, j--);
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) 
    {
        k %= nums.size();
        reverse(nums.begin(), nums.end());
        int i = 0, j = k-1;
        while(i < j) {
            swap(nums[i++], nums[j--]);
        }
        i = k, j = nums.size()-1;
        while(i < j) {
            swap(nums[i++], nums[j--]);
        }
    }
};